llama_index/vector_stores/typesense.py

```diff
@@ -0,0 +1,257 @@
+"""Typesense Vector store index.
+
+An index that that is built on top of an existing vector store.
+
+"""
+
+import logging
+from typing import Any, Callable, List, Optional, cast
+
+from llama_index import utils
+from llama_index.data_structs.node import Node
+from llama_index.vector_stores.types import (
+    MetadataFilters,
+    NodeWithEmbedding,
+    VectorStore,
+    VectorStoreQuery,
+    VectorStoreQueryMode,
+    VectorStoreQueryResult,
+)
+from llama_index.vector_stores.utils import (
+    DEFAULT_TEXT_KEY,
+    metadata_dict_to_node,
+    node_to_metadata_dict,
+)
+
+_logger = logging.getLogger(__name__)
+
+DEFAULT_COLLECTION_NAME = "default_collection"
+DEFAULT_BATCH_SIZE = 100
+DEFAULT_METADATA_KEY = "metadata"
+
+
+class TypesenseVectorStore(VectorStore):
+    """Typesense Vector Store.
+
+    In this vector store, embeddings and docs are stored within a
+    Typesense index.
+
+    During query time, the index uses Typesense to query for the top
+    k most similar nodes.
+
+    Args:
+        client (Any): Typesense client
+        tokenizer (Optional[Callable[[str], List]]): tokenizer function.
+
+    """
+
+    stores_text: bool = True
```

Comment:
since `is_embedding_query` is set to the default value of True, this means that we always generate an embedding regardless of whether or not the user is using text search right? 

if we set `is_embedding_query = False`, then we leave the burden to the user to generate an embedding from the model (like they would manually initialize an openai embedding model), and they can set `query_bundle.embedding = ...` before passing it to the vector store query engine

Reply:
Fixed